packages/tx-construction/src/createTransactionInternals.ts

```diff
@@ -14,34 +15,56 @@ export type CreateTxInternalsProps = {
   requiredExtraSignatures?: Crypto.Ed25519KeyHashHex[];
 };
 
-export const createTransactionInternals = async ({
+/** Returns transaction body based on the information available before input selection takes place */
+export const createPreInputSelectionTxBody = ({
   auxiliaryData,
   withdrawals,
   certificates,
   validityInterval,
-  inputSelection,
   collaterals,
   mint,
   scriptIntegrityHash,
-  requiredExtraSignatures
-}: CreateTxInternalsProps): Promise<Cardano.TxBodyWithHash> => {
+  requiredExtraSignatures,
+  outputs
+}: Omit<CreateTxInternalsProps, 'inputSelection'> & { outputs?: Cardano.TxOut[] }): {
+  body: TxBodyPreInputSelection;
+  auxiliaryData?: Cardano.AuxiliaryData;
+} => {
+  const body: TxBodyPreInputSelection = {
+    auxiliaryDataHash: auxiliaryData ? Cardano.computeAuxiliaryDataHash(auxiliaryData) : undefined,
+    certificates,
+    mint,
+    outputs: outputs || [],
+    requiredExtraSignatures,
+    scriptIntegrityHash,
+    validityInterval,
+    ...(withdrawals?.length && { withdrawals }),
+    ...(collaterals && { collaterals: [...collaterals] })
+  };
+
+  return {
+    auxiliaryData,
+    body
```

Comment:
I would have defined the `body` directly here... but just because I'm a body-less functions fanatic 😄

Reply:
done here: [ec76f62](https://github.com/input-output-hk/cardano-js-sdk/pull/1099/commits/ec76f62e8dff3be6334bd8bac694f380c9061544)